client/tests/integration/auto_register_account/on_transfer.rs

```diff
@@ -0,0 +1,237 @@
+//! A new account should be automatically registered when being a destination of a transfer
+
+use iroha_client::data_model::prelude::*;
+use iroha_sample_params::{alias::Alias, SampleParams};
+use serde_json::json;
+use test_network::*;
+
+/// A new account should be automatically registered when being a destination of a transfer of numeric asset e.g. "rose"
+///
+/// # Scenario
+///
+/// 0. alice@wonderland: domain owner
+/// 0. alice -> new carol ... ok (domain owner)
+/// 0. carol -> new dave ... err (permission denied)
+/// 0. grant `CanRegisterAccountInDomain` permission to carol
+/// 0. carol -> new dave ... ok (authorized)
+#[test]
+fn asset_numeric() {
```

Comment:
This transfer permissions concerns me a bit, like regular user i probably don't care if person i' trying to transfer assets is registered or not.

Can we do the following?
Transfer to any address is always allowed (what about mint and burn?), but to actually make use of this assets (e.g. transfer them to someone else) you have to register an account.
Thus by registering account you claim ownership of this assets.
Since accounts are identified by public key only owner of the private key can register the account.

Reply:
You have a point, I will consider it. If this discussion is controversial or things get too complicated, we may split the 2nd commit into a separate PR

Reply:
I think the point is the domain membership screening.
How should a new account request and be allowed to join a domain?
In terms of authority, especially in a "private domain", there is a large gap between transferring one's own asset and registering accounts within the domain.
So the admission process should be managed by a different authority than the transaction sender, as you pointed out.

One idea is that each domain has its own admission policy as a trigger that, in response to an account creation event, "activates" the new account (as a personal or a shared account).
Admission policies may vary from domain to domain e.g.

- No screening
- Wait for approval of the domain owner
- Wait for approvals of m of n domain members

In any case, as I implied, I'm going to create a new issue on this and remove the 2nd commit from this PR

Reply:
- #4426

Reply:
We seem to be in agreement that there are two types of account registration, so to speak, provisional and full registration.

Is it your opinion that the 2nd commit of this PR is acceptable, but its single registration makes an account equivalent to fully registered one, so subsequent PRs should weaken the current single registration to provisional one and newly implement the full registration?

If so, does provisional registration require permission?
If it cannot be executed without the same permission as full registration, then there would be no motive for them to split into two.
Or do we need dedicated permission for provisional registration?